packages/protect/src/lib/protect.test.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ *
+ * Copyright © 2025 Ping Identity Corporation. All right reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ *
+ */
+
+import FRStep from '@forgerock/javascript-sdk/src/fr-auth/fr-step';
+import { protect } from './protect.js';
+import {
+  noProtectType,
+  standardPingProtectEvaluationStep,
+  standardPingProtectInitializeStep,
+} from './protect.mock.data.js';
+import { ProtectConfig, Protect } from './protect.types.js';
+import { CallbackType, HiddenValueCallback } from '@forgerock/javascript-sdk';
+
+describe('protect', () => {
+  const config: ProtectConfig = {
+    envId: '12345',
+    consoleLogEnabled: true,
+    deviceAttributesToIgnore: ['userAgent'],
+    lazyMetadata: false,
+    behavioralDataCollection: true,
+    deviceKeyRsyncIntervals: 14,
+    enableTrust: false,
+    disableTags: false,
+    disableHub: false,
+  };
+  let protectAPI: Protect | undefined;
```

Comment:
I'd like to avoid relying on side-effects within the `beforeEach` and `it` functions assigning values outside of their own function blocks. Can we initialize `protect` in each `it` block?

Reply:
```ts
const protectAPI: Protect;

beforeEach(() => {
  protect = await protect(options);
});

afterEach(() => {
  protect = null;
})
```

I think this is what we would have to do, right? theres no way to not have the variable assignment to use in other child scopes.

I don't think we need all of the expect/asserts inside of the `getProtect` function, it would be little redundant to do this on every creation of the protect api.

We can just do this once in it's own `it` block.